ydb/library/yql/minikql/comp_nodes/mkql_block_map_join.cpp

```diff
@@ -267,6 +269,154 @@ using TState = TBlockJoinState<RightRequired>;
+            }
+            if (blockState.IsNotFull() && !blockState.IsFinished()) {
+                switch (Flow_->FetchValues(ctx, fields)) {
+                case EFetchResult::Yield:
+                    return EFetchResult::Yield;
+                case EFetchResult::One:
+                    blockState.Reset();
+                    continue;
+                case EFetchResult::Finish:
+                    blockState.Finish();
+                    break;
+                }
+                // Leave the loop, if no values left in the flow.
+                Y_DEBUG_ABORT_UNLESS(blockState.IsFinished());
+                break;
+            }
+            break;
+        } while(true);
+
+        if (blockState.IsEmpty()) {
+            return EFetchResult::Finish;
+        }
+        blockState.MakeBlocks(ctx.HolderFactory);
+        const auto sliceSize = blockState.Slice();
+
+        for (size_t i = 0; i < ResultJoinItems_.size(); i++) {
+            if (const auto out = output[i]) {
+                *out = blockState.Get(sliceSize, ctx.HolderFactory, i);
+            }
+        }
+
+        return EFetchResult::One;
+    }
+
+private:
+    void RegisterDependencies() const final {
+        if (const auto flow = this->FlowDependsOn(Flow_))
+            this->DependsOn(flow, Dict_);
+    }
+
+    void MakeState(TComputationContext& ctx, NUdf::TUnboxedValue& state) const {
+        state = ctx.HolderFactory.Create<TState>(ctx, LeftFlowItems_, ResultJoinItems_, ctx.WideFields.data() + WideFieldsIndex_);
+    }
+
+    TState& GetState(NUdf::TUnboxedValue& state, TComputationContext& ctx) const {
+        if (!state.HasValue()) {
+            MakeState(ctx, state);
+        }
+        return *static_cast<TState*>(state.AsBoxed().Get());
+    }
+
+    class TIterator : public TComputationValue<TIterator> {
+    using TBase = TComputationValue<TIterator>;
+        NUdf::TUnboxedValue List_;
+        NUdf::TUnboxedValue Iterator_;
+        NUdf::TUnboxedValue Current_;
+
+    public:
+        TIterator(TMemoryUsageInfo* memInfo)
+            : TBase(memInfo)
+            , List_(NUdf::TUnboxedValue::Invalid())
+            , Iterator_(NUdf::TUnboxedValue::Invalid())
+            , Current_(NUdf::TUnboxedValue::Invalid())
+        {}
+
+        inline explicit operator bool() const { return Iterator_.HasValue(); }
+        void Reset(const NUdf::TUnboxedValue&& list) {
+            List_ = std::move(list);
+            Iterator_ = List_.GetListIterator();
+        }
+        bool Next(NUdf::TUnboxedValue& item) {
+            const auto found = Iterator_.Next(Current_);
+            item = Current_;
+            return found;
+        }
+    };
+
+    void MakeIterator(TComputationContext& ctx, NUdf::TUnboxedValue& iterator) const {
+        iterator = ctx.HolderFactory.Create<TIterator>();
+    }
+
+    TIterator& GetIterator(NUdf::TUnboxedValue& iterator, TComputationContext& ctx) const {
+        if (!iterator.HasValue()) {
```

Comment:
the same

Reply:
[Fixed](https://github.com/ydb-platform/ydb/pull/8501/commits/9ec60d973e656577d21cfcf1d169f17c578c7480).